src/radiation/radiation_system.hpp

```diff
@@ -88,6 +89,29 @@ struct RadPressureResult {
 	double S;		       // maximum wavespeed for the radiation system
 };
 
+template <typename problem_t> struct NewtonIterationResult {
+	double Egas;									      // gas internal energy
+	double T_gas;									      // gas temperature
+	double T_d;									      // dust temperature
+	quokka::valarray<double, Physics_Traits<problem_t>::nGroups> EradVec;		      // radiation energy density
+	quokka::valarray<double, Physics_Traits<problem_t>::nGroups> kappaPVec;		      // Planck mean opacity
+	quokka::valarray<double, Physics_Traits<problem_t>::nGroups> kappaEVec;		      // energy mean opacity
+	quokka::valarray<double, Physics_Traits<problem_t>::nGroups> kappaFVec;		      // flux mean opacity
+	quokka::valarray<double, Physics_Traits<problem_t>::nGroups> work;		      // work term
+	amrex::GpuArray<double, Physics_Traits<problem_t>::nGroups> delta_nu_kappa_B_at_edge; // Delta (nu * kappa_B * B)
+};
+
+// A struct to hold the results of ComputeJacobianForPureGas or ComputeJacobianForGasAndDust
+template <typename problem_t> struct JacobianResult {
+	double J00;	   // (0, 0) component of the Jacobian matrix
```

Comment:
What variables does this correspond to?

Reply:
This `JacobianResult` can be used to hold any Jacobian matrix of similar size, so it depends on the problem we are solving. For pure gas or dust in the coupled regime, J00 is d F_G / d Egas. For dust in the decoupled regime, it's d F_G / d T_d. So, I wouldn't add comments to explain them here, but instead, explain all elements inside each `ComputeJacobian` functions.